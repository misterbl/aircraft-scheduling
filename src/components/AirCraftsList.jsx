import React from "react";
import AirCraft from "./Aircraft";

const AirCraftsList = ({
  aircraftList,
  totalAircraftUse,
  onSelectAircraft
}) => (
  <div className="container mw-250">
    <h3>Aircrafts</h3>
    <ul className="d-flex flex-column">
      {aircraftList.map(aircraft => (
        <AirCraft
          key={aircraft.ident}
          aircraft={aircraft}
          totalAircraftUse={totalAircraftUse}
          onSelectAircraft={onSelectAircraft}
        />
      ))}
    </ul>
  </div>
);

export default AirCraftsList;
