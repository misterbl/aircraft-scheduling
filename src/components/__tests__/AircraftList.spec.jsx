import * as React from "react";
import { shallow } from "enzyme";
import AirCraftsList from "../AirCraftsList";

describe("AirCraftsList", () => {
  const props = {
    aircraftList: [{ ident: "1234" }],
    totalAircraftUse: jest.fn(() => 42),
    onSelectAircraft: jest.fn()
  };
  const wrapper = shallow(<AirCraftsList {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls onSelectAircraft onSelectAircraft click", () => {
    const airCraft = wrapper.find("AirCraft");
    airCraft.simulate("selectAircraft");
    expect(props.onSelectAircraft).toHaveBeenCalled();
  });
});
