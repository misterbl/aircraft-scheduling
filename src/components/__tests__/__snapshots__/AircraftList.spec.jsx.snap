// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirCraftsList matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AirCraftsList
    aircraftList={
      Array [
        Object {
          "ident": "1234",
        },
      ]
    }
    onSelectAircraft={[MockFunction]}
    totalAircraftUse={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Aircrafts
        </h3>,
        <ul
          className="d-flex flex-column"
        >
          <AirCraft
            aircraft={
              Object {
                "ident": "1234",
              }
            }
            onSelectAircraft={[MockFunction]}
            totalAircraftUse={[MockFunction]}
          />
        </ul>,
      ],
      "className": "container mw-250",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Aircrafts",
        },
        "ref": null,
        "rendered": "Aircrafts",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <AirCraft
              aircraft={
                Object {
                  "ident": "1234",
                }
              }
              onSelectAircraft={[MockFunction]}
              totalAircraftUse={[MockFunction]}
            />,
          ],
          "className": "d-flex flex-column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1234",
            "nodeType": "function",
            "props": Object {
              "aircraft": Object {
                "ident": "1234",
              },
              "onSelectAircraft": [MockFunction],
              "totalAircraftUse": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Aircrafts
          </h3>,
          <ul
            className="d-flex flex-column"
          >
            <AirCraft
              aircraft={
                Object {
                  "ident": "1234",
                }
              }
              onSelectAircraft={[MockFunction]}
              totalAircraftUse={[MockFunction]}
            />
          </ul>,
        ],
        "className": "container mw-250",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Aircrafts",
          },
          "ref": null,
          "rendered": "Aircrafts",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <AirCraft
                aircraft={
                  Object {
                    "ident": "1234",
                  }
                }
                onSelectAircraft={[MockFunction]}
                totalAircraftUse={[MockFunction]}
              />,
            ],
            "className": "d-flex flex-column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1234",
              "nodeType": "function",
              "props": Object {
                "aircraft": Object {
                  "ident": "1234",
                },
                "onSelectAircraft": [MockFunction],
                "totalAircraftUse": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
